src/commands/ProtectionsCommands.ts

```diff
@@ -33,6 +33,38 @@ export async function execEnableProtection(roomId: string, event: any, mjolnir:
     }
 }
 
+export async function execSetProtection(roomId: string, event: any, mjolnir: Mjolnir, parts: string[]) {
+    const [protectionName, ...settingParts] = parts[2].split(".");
+    const protection = PROTECTIONS[protectionName];
+    if (protection !== undefined) {
+        const defaultSettings = protection.factory().settings
+        const settingName = settingParts[0];
+        const value = parts[3];
+
+        let newSettings: { [settingName: string]: any } = {}
+        let settingInt = 0;
+
+        let message = "did nothing";
+
+        if (defaultSettings[settingName] === undefined) {
+            message = `Unknown setting ${settingName}`;
+        } else if (typeof(defaultSettings[settingName]) === 'number'
+                && !isNaN(settingInt = Number(value))) {
+            newSettings[settingName] = settingInt;
+        } else {
+            message = `Type mismatch (should be ${typeof(defaultSettings[settingName])})`;
```

Comment:
I'd prefer bailing out immediately. Less accident-prone for the day someone else tinkers with that code.